action.yml

```
@@ -0,0 +1,140 @@
+name: Gauge release
+description: Gauge if should do a patch release
+
+inputs:
+  latest_local_sha:
+    description: The latest local sha
+    required: true
+    type: string
+
+outputs:
+  do_release:
+    description: Whether to do a release
+    value: ${{ steps.gauge-release.outputs.do_release }}
+  next_tag:
+    description: The patch tag to release
+    value: ${{ steps.gauge-release.outputs.next_tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Gauge release
+      id: gauge-release
+      shell: bash
+      env:
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        GITHUB_REF_NAME: ${{ github.ref_name }}
+        GITHUB_SHA: ${{ github.sha }}
+        LATEST_LOCAL_SHA: ${{ inputs.latest_local_sha }}
+      run: |
+        DO_RELEASE=1
+
+        # Double check that LATEST_LOCAL_SHA matches GITHUB_SHA
+        if [[ $DO_RELEASE == "1" ]]; then
+          if [[ $LATEST_LOCAL_SHA != $GITHUB_SHA ]]; then
+            echo "LATEST_LOCAL_SHA $LATEST_LOCAL_SHA does not match GITHUB_SHA $GITHUB_SHA"
+            DO_RELEASE=0
+          fi
+        fi
+
+        # Must be on a minor branch to do a patch release
+        if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0-9]+$ ]]; then
+          echo "Not patch releasing because not on a minor branch"
+          DO_RELEASE=0
+        fi
+
+        # Validate that this commit is that latest commit for the branch using GitHub API
+        # We need to check this in case re-rerunning an old job and there have been new commits since
+        if [[ $DO_RELEASE == "1" ]]; then
+          # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28
+          RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+          -X GET "https://api.github.com/repos/${GITHUB_REPOSITORY}/commits?sha=${GITHUB_REF_NAME}&per_page=1" \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ github.token }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          )
+          if [[ $RESP_CODE != "200" ]]; then
+            echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+            exit 1
+          fi
+          LATEST_REMOTE_SHA=$(jq -r '.[0].sha' __response.json)
+          echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
```

```suggestion
          echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
          echo "LATEST_LOCAL_SHA is $LATEST_LOCAL_SHA"
```
You removed this but we're not echoing it anywhere else, so we either need it here or above as I've suggested in another comment.

---

Updated